packages/components/src/components/domainproperties/dataset/actions.ts

```diff
@@ -23,40 +31,92 @@ import {
-            { label: 'Cohort3', value: 3 },
-        ],
-    };
-};
-
-export const fetchVisitDateColumns = async () => {
-    // TODO: Keeping this action until next story in which visitDateColumns will be pulled from state change (for date fields) in the Domain Form.
-    return {
-        visitDateColumns: [
-            { label: 'Date', value: 'date' },
-            { label: 'Arrival Date', value: 'arrivalDate' },
-        ],
-    };
-};
+export function fetchCategories(): Promise<List<Option>> {
+    return new Promise((resolve, reject) => {
+        selectRows({
+            saveInSession: true,
+            schemaName: 'study',
+            sql: 'SELECT DISTINCT CategoryId.Label, CategoryId.RowId FROM DataSets',
+        })
+            .then(data => {
+                const models = fromJS(data.models[data.key]);
+                let categories = List<Option>();
+
+                data.orderedModels[data.key].forEach(modelKey => {
+                    const row = models.get(modelKey);
+                    const value = row.getIn(['Label', 'value']);
+                    const label = row.getIn(['Label', 'value']);
+
+                    categories = categories.push({ value, label });
+                });
+
+                resolve(categories);
+            })
+            .catch(response => {
+                reject(response.message);
+            });
+    });
+}
+
+export function fetchVisitDateColumns(domain: DomainDesign): List<Option> {
+    let visitDateColumns = List<Option>();
+
+    visitDateColumns = visitDateColumns.push({ value: 'date', label: 'date' });
+
+    domain.fields.map(field => {
+        if (field.rangeURI.endsWith('dateTime')) {
+            visitDateColumns = visitDateColumns.push({ value: field.name, label: field.name });
+        }
+    });
+
+    return visitDateColumns;
+}
+
+export function fetchAdditionalKeyFields(domain: DomainDesign): List<Option> {
+    let additionalKeyFields = List<Option>();
+
+    if (getServerContext().moduleContext.study.timepointType !== 'VISIT') {
+        additionalKeyFields = additionalKeyFields.push({ value: TIME_KEY_FIELD_KEY, label: TIME_KEY_FIELD_DISPLAY });
+    }
+
+    domain.fields.map(field => {
+        additionalKeyFields = additionalKeyFields.push({ value: field.name, label: field.name });
+    });
+
+    return additionalKeyFields;
+}
+
+export function fetchCohorts(): Promise<List<Option>> {
+    return new Promise((resolve, reject) => {
+        selectRows({
+            saveInSession: true,
+            schemaName: 'study',
+            queryName: 'Cohort',
+        })
+            .then(data => {
+                const models = fromJS(data.models[data.key]);
+                let cohorts = List<Option>();
+
+                data.orderedModels[data.key].forEach(modelKey => {
+                    const row = models.get(modelKey);
+                    const value = row.getIn(['rowid', 'value']);
+                    const label = row.getIn(['label', 'value']);
```

Comment:
might want to double check this on MSSQL and PG for column label casing differences (maybe you already did), i.e. rowid vs RowId.

Reply:
checked, works for both